src/components/checkbox/checkbox.scss

```diff
@@ -39,6 +39,20 @@
                 }
             }
         }
+
+        &.mdc-checkbox--anim-checked-unchecked .mdc-checkbox__native-control:enabled~.mdc-checkbox__background,
+        &.mdc-checkbox--anim-indeterminate-unchecked .mdc-checkbox__native-control:enabled~.mdc-checkbox__background {
+            /*
+            don't show the checkmark when transitioning from mixedmark (indeterminate)
+            to unchecked. Expected behavior is to show indeterminate -> unchecked
+            without showing the checkmark during the transition.
+            The checked->unchecked selector is used because the checkbox transitions
+            from indeterminate->checked->unchecked when the user clicks a
+            checkbox when it is in the indeterminate state.
+            note: transition indeterminate->checked is still animated on purpose
+            */
+            animation-name: none;
+        }
```

Comment:
A clip of the annoying effect when this CSS override is left out:
https://user-images.githubusercontent.com/435885/149826966-7484174b-6230-422f-9876-0bde6431aba7.mov

Reply:
@Kiarokh maybe setting animation-name to none like this can work for your loading animation issue too?

Reply:
@anderssonjohan not sure what the discussion is about here. I didn't read the comments, as I feel I'm not the right person to have opinions here. But I suggest that you leave all animation problems out of this PR. When this is merged, I'll go through the checkbox component and fix/improve/disable the animations in a separate PR. :)

Reply:
@Kiarokh it was just a tip for you if your change of keyframes didn't work out. Removing the animation name may be another solution for you, that was all :)